lib/fileset/default.nix

```diff
@@ -187,6 +188,24 @@ If a directory does not recursively contain any file, it is omitted from the sto
         filter = sourceFilter;
       };
 
+  /*
+  Create a file set from a filtered local source as produced by the `lib.sources` functions.
+  This does not import anything into the store.
+
+  Type:
+    fromSource :: SourceLike -> FileSet
+
+  Example:
+    fromSource (lib.sources.cleanSource ./.)
+  */
+  fromSource = source:
+    if ! source._isLibCleanSourceWith or false || ! source ? origSrc || ! source ? filter then
+      throw "lib.fileset.fromSource: Expected the argument to be a value produced from `lib.sources`, but got a ${typeOf source} instead."
+    else if ! isPath source.origSrc then
+      throw "lib.fileset.fromSource: Expected the argument to be source-like value of a local path."
```

Comment:
What's a source-like value? What's a source?

Reply:
"Very" good question unfortunately. Basically a path value or the "source attrset" returned by the `lib.sources` functions. Can we write introductions for sublibraries yet? It'd be great to have a sensible place to document the unique types of each sublibrary.

Reply:
> Can we write introductions for sublibraries yet? 

[Almost](https://github.com/nix-community/nixdoc/pull/70)